src/net/http/chunk.clj

```diff
@@ -147,7 +147,7 @@
     [this ftr [^ChannelHandlerContext ctx ^Channel ch]]
     (if (or (nil? ftr) (f/complete? ftr))
       (a/take! ch (write-listener-callback this ctx))
-      (do (a/close! ch) (a/go (while (a/<! ch)))))))
+      (do (a/close! ch) (drain ch buf/ensure-released)))))
```

Comment:
We're unlikely to ever `drain` without needing to `a/close!`. I'd suggest moving the `a/close!` in `drain` directly.